ecosystem/typescript/sdk/src/utils/memoize-decorator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Credits to https://github.com/darrylhodgins/typescript-memoize
+ */
+
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+
+interface MemoizeArgs {
+  expiring?: number;
+  hashFunction?: boolean | ((...args: any[]) => any);
+  tags?: string[];
+}
+
+export function Memoize(args?: MemoizeArgs | MemoizeArgs["hashFunction"]) {
+  let hashFunction: MemoizeArgs["hashFunction"];
+  let duration: MemoizeArgs["expiring"];
+  let tags: MemoizeArgs["tags"];
```

Comment:
Why do this instead of just using `args.field` directly?

Reply:
Are you talking about `MemoizeArgs["hashFunction"]` vs `MemoizeArgs.hashFunction`?  `MemoizeArgs` is a an interface, you can only use bracket notation not the dot notation